src/Microsoft.SqlTools.ResourceProvider.Core/Contracts/FirewallRule.cs

```diff
@@ -0,0 +1,102 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ResourceProvider.Core.Contracts
+{
+    /// <summary>
+    /// A request to open up a firewall rule
+    /// </summary>
+    public class CreateFirewallRuleRequest
+    {
+        public static readonly
+            RequestType<CreateFirewallRuleParams, CreateFirewallRuleResponse> Type =
+            RequestType<CreateFirewallRuleParams, CreateFirewallRuleResponse>.Create("resource/createFirewallRule");
+    }
+
+    /// <summary>
+    /// A FirewallRule object, usable in <see cref="CreateFirewallRuleRequest"/>s and other messages
+    /// </summary>
+    public class CreateFirewallRuleParams
+    {
+        /// <summary>
+        /// Account information to use in connecting to Azure
+        /// </summary>
+        public Account Account { get; set; }
+        /// <summary>
+        /// Per-tenant oken mappings. Ideally would be set independently of this call, but for 
+        /// now this allows us to handle this
+        /// </summary>
+        public Dictionary<string,string> SecurityTokenMappings { get; set; }
```

Comment:
This will actually be a (copy/paste definitions from typescript)
```
/**
 * Azure account security token maps a tenant ID to the information returned from a request to get
 * an access token. The list of tenants correspond to the tenants in the account properties.
 */
export type AzureAccountSecurityTokenCollection = {[tenantId: string]: AzureAccountSecurityToken};

/**
 * Token returned from a request for an access token
 */
export interface AzureAccountSecurityToken {
	/**
	 * Access token, itself
	 */
	token: string;

	/**
	 * Date that the token expires on
	 */
	expiresOn: Date;

	/**
	 * Name of the resource the token is good for (ie, management.core.windows.net)
	 */
	resource: string;

	/**
	 * Type of the token (pretty much always 'Bearer')
	 */
	tokenType: string;
}
```